src/main/java/org/nuxeo/ecm/core/storage/marklogic/MarkLogicRepository.java

```diff
@@ -127,7 +133,26 @@ public static ContentSource newMarkLogicContentSource(MarkLogicRepositoryDescrip
         String dbname = StringUtils.defaultIfBlank(descriptor.dbname, DB_DEFAULT);
         String user = descriptor.user;
         String password = descriptor.password;
-        return ContentSourceFactory.newContentSource(host, port.intValue(), user, password, dbname);
+        // handle SSL
+        SecurityOptions securityOptions = null;
+        if (descriptor.sslEnabled) {
+            securityOptions = newTrustOptions();
+        }
+        return ContentSourceFactory.newContentSource(host, port.intValue(), user, password, dbname, securityOptions);
+    }
+
+    protected static SecurityOptions newTrustOptions() {
+        try {
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance("SunX509");
```

Comment:
By using `SSLContext.getDefault()` we don't need anymore the TrustManagerFactory.
Furthermore `SSLContext.getDefault()` use the default `PKIX` algorithm.